apps/print/backend/src/reports/ballots_printed_report.ts

```diff
@@ -0,0 +1,110 @@
+import { assert, assertDefined } from '@votingworks/basics';
+import { BallotsPrintedReport } from '@votingworks/ui';
+import { LogEventId, Logger } from '@votingworks/logging';
+import { Printer, renderToPdf } from '@votingworks/printing';
+import { UsbDrive } from '@votingworks/usb-drive';
+import { Exporter } from '@votingworks/backend';
+import { join } from 'node:path';
+import {
+  generateElectionBasedSubfolderName,
+  REPORT_FOLDER,
+  generateFileTimeSuffix,
+} from '@votingworks/utils';
+import { ElectionDefinition } from '@votingworks/types';
+import { Store } from '../store';
+import { PRINT_ALLOWED_EXPORT_PATTERNS } from '../globals';
+
+export function generateReportsDirectoryPath(
+  electionDefinition: ElectionDefinition
+): string {
+  return join(
+    generateElectionBasedSubfolderName(
+      electionDefinition.election,
+      electionDefinition.ballotHash
+    ),
+    REPORT_FOLDER
+  );
+}
+
+function buildBallotsPrintedReport({ store }: { store: Store }): JSX.Element {
+  const electionRecord = assertDefined(store.getElectionRecord());
+  const { electionDefinition, electionPackageHash } = electionRecord;
+
+  const printCounts = store.getBallotPrintCounts({});
+
+  return BallotsPrintedReport({
+    electionDefinition,
+    electionPackageHash,
+    printCounts,
+    generatedAtTime: new Date(),
+  });
+}
+
+export async function printBallotsPrintedReport({
+  printer,
+  logger,
+  store,
+}: {
+  printer: Printer;
+  logger: Logger;
+  store: Store;
+}): Promise<void> {
+  const report = buildBallotsPrintedReport({ store });
+
+  try {
+    const data = (await renderToPdf({ document: report })).unsafeUnwrap();
```

Comment:
nit: you can use the `Result` from `await renderToPdf(...)` to check for an error and leave it outside the try/catch.

```
const renderResult = await renderToPdf({ document: report }));
if (renderResult.isErr()) {
  const error = renderResult.err();
  // log error
}
const data = renderResult.ok();
try {
  await printer.print({ data });
  ...
}
```

It's marginally clearer this way because an error in `renderToPdf` won't report as "Error in attempting to print ballots printed report".